client/web/src/hooks/auth.ts

```diff
@@ -31,6 +31,11 @@ export default function useAuth() {
     return apiFetch("/auth", "GET")
       .then((r) => r.json())
       .then((d) => {
+        // going from a defined auth type value in the old data to
+        // undefined in the response from auth implies succesful auth
+        if (data?.authNeeded && !(d as AuthResponse).authNeeded) {
```

Comment:
Oh I think I see the confusing here with what we were threading about yesterday. I think there's two different "auth"-related sets of events:

1. When the node is connected to a tailnet; the opposite of this is when the node is disconnected from the tailnet (you have the disconnect one recorded as the deauth event right now). I'm proposing we move the connection event over to login-view.tsx.
2. When the viewing user's current auth for viewing the web client is determined. This is what we were talking about earlier in terms of bucketing the user into various groups, and might make more sense from the backend handler for the "/api/auth" endpoint. This is the event that would likely replace page views in terms of usefulness (because it'll get sent once for web client app use).

Reply:
Gotcha. So for the "node is connected to a tailnet" event we would be plugging in [here](https://github.com/tailscale/tailscale/blob/main/client/web/src/components/views/login-view.tsx#L28) then?

Reply:
Yes exactly!